// Copyright 2021 Google LLC. All Rights Reserved.
// 
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
// 
//     http://www.apache.org/licenses/LICENSE-2.0
// 
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
syntax = "proto3";

package dcl;

import "proto/connector/sdk.proto";
import "proto/empty.proto";

enum ApigeeEnvironmentStateEnum {
  ApigeeEnvironmentStateEnumNO_VALUE_DO_NOT_USE = 0;
  ApigeeEnvironmentStateEnumSTATE_UNSPECIFIED = 1;
  ApigeeEnvironmentStateEnumCREATING = 2;
  ApigeeEnvironmentStateEnumACTIVE = 3;
  ApigeeEnvironmentStateEnumDELETING = 4;
}

message ApigeeEnvironment {
  string name = 1;
  string description = 2;
  int64 created_at = 3;
  int64 last_modified_at = 4;
  ApigeeEnvironmentProperties properties = 5;
  string display_name = 6;
  ApigeeEnvironmentStateEnum state = 7;
  string organization = 8;
}

message ApigeeEnvironmentProperties {
  repeated ApigeeEnvironmentPropertiesProperty property = 1;
}

message ApigeeEnvironmentPropertiesProperty {
  string name = 1;
  string value = 2;
}

message ApplyApigeeEnvironmentRequest {
  ApigeeEnvironment resource = 1;
  repeated LifecycleDirective lifecycle_directives = 2;
  string service_account_file = 3;
}

message DeleteApigeeEnvironmentRequest {
  string service_account_file = 1;
  ApigeeEnvironment resource = 2;
}

message ListApigeeEnvironmentRequest {
  string service_account_file = 1;
  string Organization = 2;
}

message ListApigeeEnvironmentResponse {
  repeated ApigeeEnvironment items = 1;
}

service ApigeeEnvironmentService {
  rpc ApplyApigeeEnvironment(ApplyApigeeEnvironmentRequest) returns (ApigeeEnvironment);
  rpc DeleteApigeeEnvironment(DeleteApigeeEnvironmentRequest) returns (google.protobuf.Empty);
  rpc ListApigeeEnvironment(ListApigeeEnvironmentRequest) returns (ListApigeeEnvironmentResponse);
}
